controllers/database_secret_handler.go

```diff
@@ -0,0 +1,160 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package controllers
+
+import (
+	"context"
+	kciv1alpha1 "github.com/kloeckner-i/db-operator/api/v1alpha1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/client-go/util/workqueue"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	DbSecretAnnotation = "db-operator/database"
+)
+
+/* ------ Secret Event Handler ------ */
+type secretEventHandler struct {
+	client.Client
+}
+
+func (e *secretEventHandler) Update(evt event.UpdateEvent, q workqueue.RateLimitingInterface) {
+
+	logrus.Info("Secret Update Event processing started")
+
+	switch v := evt.ObjectNew.(type) {
+
+	default:
+		logrus.Error("Secret Update Event error! Unknown object: type=", v.GetObjectKind(), ", name=", evt.ObjectNew.GetNamespace(), "/", evt.ObjectNew.GetName())
+		return
+
+	case *corev1.Secret:
+		secretNew := evt.ObjectNew.(*corev1.Secret)
+		// find database object
+
+		// only annotated secrets are watched
+		annotations := secretNew.ObjectMeta.GetAnnotations()
+		dbcrName := annotations[DbSecretAnnotation]
+
+		if len(dbcrName) == 0 {
+			logrus.Error("Secret Update Event error! Annotation '", DbSecretAnnotation, "' value is empty.")
+			return
+		}
+		// send Database reconcile event
+		logrus.Info("Database Secret Data has been changed and related Database resource will be reconciled: secret=", secretNew.Namespace, "/", secretNew.Name, ", database=", dbcrName)
+		q.Add(reconcile.Request{NamespacedName: types.NamespacedName{
+			Namespace: secretNew.GetNamespace(),
+			Name:      dbcrName,
+		}})
+	}
+}
+
+func (e *secretEventHandler) Delete(event.DeleteEvent, workqueue.RateLimitingInterface) {
+	logrus.Error("secretEventHandler.Delete(...) event has been FIRED but NOT implemented!")
+}
+func (e *secretEventHandler) Generic(event.GenericEvent, workqueue.RateLimitingInterface) {
+	logrus.Error("secretEventHandler.Generic(...) event has been FIRED but NOT implemented!")
+}
+func (e *secretEventHandler) Create(evt event.CreateEvent, q workqueue.RateLimitingInterface) {
+	logrus.Error("secretEventHandler.Create(...) event has been FIRED but NOT implemented!")
+}
+
+/* ------ Event Filter Functions ------ */
+
+func getDatabasesBySecret(c client.Client, secret types.NamespacedName) ([]kciv1alpha1.Database, []string, error) {
```

Comment:
think this function is not needed anymore, or?

Reply:
removed